base/shell.jl

```diff
@@ -16,8 +16,20 @@ function rstrip_shell(s::AbstractString)
     SubString(s, 1, 0)
 end
 
+# trailing double-backslashes in custom string literals are only parsed as single backslashes
+# as to still allow escaping the delimiter. This is needed to parse e.g. `\\` correctly
+function fix_trailing_backslashes(s::AbstractString)
+    i = lastindex(s)
+    while i > 0 && s[i] == '\\'
+        s *= '\\'
+        i = prevind(s, i)
+    end
+    return s
+end
+
 function shell_parse(str::AbstractString, interpolate::Bool=true;
                      special::AbstractString="", filename="none")
+    str = fix_trailing_backslashes(str)
```

Comment:
```suggestion
    str = escape_raw_string(str)
```

Reply:
I don't think that's quite what we want here, since we still want `` `\"` `` to be equivalent to `` `'"'` ``, not `` `'\"'` ``.

Reply:
Ah, perhaps `escape_raw_string` needs an extra argument for which character is the delimiter (` vs "). The name "fix" is just really bad, since it makes it sound like something was broken, but really this is just better with the re-escape call there.

Reply:
We also don't really want to escape `` ` `` here though, so I am not sure this should be in `escape_raw_string`. Would you be okay with just renaming this little helper function to `escape_trailing_backslash`?

Reply:
Does that look ok to you now?